src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -140,9 +141,16 @@ public function createListFromLoader(ChoiceLoaderInterface $loader, $value = nul
     public function createView(ChoiceListInterface $list, $preferredChoices = null, $label = null, $index = null, $groupBy = null, $attr = null)
     {
         // Backwards compatibility
-        if ($list instanceof LegacyChoiceListInterface && null === $preferredChoices
+        if ($list instanceof LegacyChoiceListInterface && empty($preferredChoices)
             && null === $label && null === $index && null === $groupBy && null === $attr) {
-            return new ChoiceListView($list->getRemainingViews(), $list->getPreferredViews());
+            $mapToNonLegacyChoiceView = function (LegacyChoiceView $choiceView) {
+                return new ChoiceView($choiceView->label, $choiceView->value, $choiceView->data);
```

Comment:
I think a more efficient solution would be to simply define the "attr" property dynamically:

```php
$choiceView->attr = array();
```

After all, this method is called pretty often.